### PR TITLE
A15 - Priority Queue

### PR DESCRIPTION
**Pull request by vikrampg1710-tmm**

### CODE REVIEW

File: Assignments/Program.cs [hunk 1]
```diff
@@ -1,3 +1,82 @@
...
+
+   #region Priority Queue Class -------------------------------------------------------------------
+   /// <summary>
+   /// Represents a collection of items that have a value and a priority. 
+   /// On dequeue, the item with the lowest priority value is removed.
+   /// </summary>
+   /// <typeparam name="T"></typeparam>
+   public class PriorityQueue<T> where T : IComparable<T> {
+      #region Propeties ---------------------------------------------
+      readonly List<T> queue = new ();
```
Comment (ravin-imm) [inline Assignments/Program.cs]: Naming convention not followed.

File: Assignments/Program.cs [hunk 2]
```diff
@@ -1,3 +1,82 @@
...
+         else {
+            var item = queue[0];
+            queue[0] = queue[^1];
+            queue.RemoveAt (queue.Count - 1);
+            ShiftDown (0);
+            return item;
+         }
+
+         ///<summary>Rearranges the queue elements based on the priority value by shifting down the elements</summary>
+         void ShiftDown (int parentIndex) {
```
Comment (ravin-imm) [inline Assignments/Program.cs]: The method names should be **SiftDown** and **SiftUp**.

File: Assignments/Program.cs [hunk 3]
```diff
@@ -1,3 +1,82 @@
...
+      readonly List<T> queue = new ();
+      public bool IsEmpty => queue.Count == 0;
+      #endregion
+
+      #region Methods -----------------------------------------------
+      /// <summary>
+      /// Removes and returns the minimal element from the PriorityQueue<typeparamref name="T"/> - that is, the element with the lowest priority value.
+      /// </summary>
+      /// <returns>Returns the element with the lowest priority value from the queue</returns>
+      /// <exception cref="Exception"></exception>
```
Comment (ravin-imm) [inline Assignments/Program.cs]: Format the comments

File: Assignments/Program.cs [hunk 4]
```diff
@@ -1,3 +1,82 @@
...
+            var item = queue[0];
+            queue[0] = queue[^1];
+            queue.RemoveAt (queue.Count - 1);
+            ShiftDown (0);
+            return item;
+         }
+
+         ///<summary>Rearranges the queue elements based on the priority value by shifting down the elements</summary>
+         void ShiftDown (int parentIndex) {
+            int child1_Index = parentIndex * 2 + 1;
```
Comment (ravin-imm) [inline Assignments/Program.cs]: Try to use short words for variable names.

**Review by ravin-imm — CHANGES_REQUESTED** [review]

File: Assignments/Program.cs (hunk 3 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: Can you please specify the error?

File: Assignments/Program.cs (hunk 1 above)
Comment (vikrampg1710-tmm) [reply Assignments/Program.cs]: The field member should have prefixed with 'm'.  I will modify it.

File: Assignments/Program.cs (hunk 3 above)
Comment (ravin-imm) [reply Assignments/Program.cs]: @vikrampg1710-tmm: The start and the end tags should not be in separate lines. The above text can be:
```
      /// <summary>Removes and returns the minimal element from the PriorityQueue<typeparamref name="T"/> - that is, the element with the lowest priority value.</summary>
      /// <returns>Returns the element with the lowest priority value from the queue</returns>
      /// <exception cref="Exception"></exception>
```

**Review by ravin-imm — APPROVED** [review]

**Pull request closed without merging**